### PR TITLE
s3-bucket-ssl-requests-only

### PR DESCRIPTION
**Pull request by mlevit**

### CODE REVIEW

File: auto_remediate/config_rules.py [hunk 1]
```diff
@@ -37,11 +61,11 @@ def rds_instance_public_access_check(self, resource_id):
             self.logging.error("Could not describe RDS DB Instances.")
             return False
         else:
-            for instance in response.get("DBInstances"):
-                if resource_id == instance.get("DbiResourceId"):
+            for instance in response["DBInstances"]:
```
Comment (jihygk) [inline auto_remediate/config_rules.py:65]: #34

File: auto_remediate/config_rules.py [hunk 2]
```diff
@@ -87,3 +111,145 @@ def s3_bucket_server_side_encryption_enabled(self, resource_id):
...
+                try:
+                    self.client_s3.put_bucket_policy(
+                        Bucket=resource_id, Policy=json.dumps(policy)
+                    )
+                    self.logging.info(
+                        f"Set SSL requests only policy to S3 Bucket '{resource_id}'."
+                    )
+                    return True
+                except ClientError as error:
+                    if error.response["Error"]["Code"] == "AccessDenied":
```
Comment (jihygk) [inline auto_remediate/config_rules.py]: refactor this duplicated logic out into function

**Review by jihygk — CHANGES_REQUESTED** [review]

File: auto_remediate/config_rules.py (hunk 1 above)
Comment (mlevit) [reply auto_remediate/config_rules.py:65]: 93657b1

File: auto_remediate/config_rules.py (hunk 2 above)
Comment (mlevit) [reply auto_remediate/config_rules.py]: 13b6292

**Review by jihygk — APPROVED** [review]

**Pull request merged**